packages/near-fast-auth-signer/src/lib/useAuthState.ts

```diff
@@ -10,90 +11,95 @@ import { redirectWithError, safeGetLocalStorage } from '../utils';
+
+      if (accountsList.length === 0) {
+        return false;
+      }
+
+      window.fastAuthController = new FastAuthController({
+        accountId: accountsList[0].accId,
+        networkId
+      });
+
+      await window.fastAuthController.setKey(new KeyPairEd25519(accountsList[0].keyPair.toString().split(':')[1]));
+      return true;
+    } catch {
+      return false;
+    }
+  } else if (isFirestoreReady && firebaseAuth.currentUser) {
+    const oidcToken = await firebaseAuth.currentUser.getIdToken();
+    const localStoreKey = await window.fastAuthController.getLocalStoreKey(`oidc_keypair_${oidcToken}`);
+
+    if (localStoreKey) {
+      const account = await window.fastAuthController.recoverAccountWithOIDCToken(oidcToken);
+      window.fastAuthController = new FastAuthController({
+        accountId: account?.accountId,
+        networkId
+      });
+      return true;
+    }
+  }
+
+  return false;
+};
+
+export const useAuthState = (skipGetKeys = false): {authenticated: AuthState} => {
+  const [authenticated, setAuthenticated] = useState<AuthState>('loading');
   const [query] = useSearchParams();
   const email = query.get('email');
   const successUrl = query.get('success_url');
   const failureUrl = query.get('failure_url');
 
-  if (webauthnUsername === undefined) {
-    return { authenticated: new Error('Please allow third party cookies') };
-  }
-
   useEffect(() => {
     const handleAuthState = async () => {
-      const controllerState = await window.fastAuthController.isSignedIn();
-      const isPasskeySupported = await isPassKeyAvailable();
-
-      if (skipGetKeys) {
-        setAuthenticated(false);
-      } else if (controllerState === true) {
-        setAuthenticated(true);
-      } else if (isPasskeySupported && (!webauthnUsername || (email && email !== webauthnUsername))) {
-        setAuthenticated(false);
-      } else if (isPasskeySupported) {
-        try {
-          const keypairs = await getKeys(webauthnUsername);
-          const accounts = await Promise.allSettled(
-            keypairs.map(async (k) => {
-              const accIds = await fetchAccountIds(k.getPublicKey().toString());
-              return accIds.map((accId) => { return { accId, keyPair: k }; });
-            })
-          );
-
-          const accountsList = accounts
-            .flatMap((result) => (result.status === 'fulfilled' ? result.value : []));
-
-          if (accountsList.length === 0) {
-            setAuthenticated(false);
-          }
-
-          window.fastAuthController = new FastAuthController({
-            accountId: accountsList[0].accId,
-            networkId
-          });
-
-          await window.fastAuthController.setKey(new KeyPairEd25519(accountsList[0].keyPair.toString().split(':')[1]));
-          setAuthenticated(true);
-        } catch {
-          setAuthenticated(false);
+      try {
+        if (!email) {
+          throw new Error('Email is required');
```

Comment:
In the original implementation of this hook, we would return the auth state retrieved from the process of `getKeys` + indexer request etc in all cases except if the email in the query params is different from the locally stored `webauthn_username`. Why are we requiring the query param of the current page to have an email address here?